openedx/core/lib/xblock_utils.py

```diff
@@ -211,6 +212,12 @@ def grade_histogram(module_id):
     return grades
 
 
+def sanitize_html_id(html_id):
+    """Template uses element_id in js function names, so can't allow dashes and colons"""
```

Comment:
Hi @robrap thanks for taking the time to review this PR.
For when ``html_id`` contained dashes as special characters there was a ``replace`` in place down at line: https://github.com/edx/edx-platform/pull/10352/files#diff-fd85edee8483bf8cc2814b8d453d12d1L303 
MIT folks, as I've been told, tend to create their courses as XML files and import them that way. When you would group, for example problems, into one folder you would get a structure like this:
```
problem group
    | - problem1
    | - problem2
  . . .
```
and that was reflected in the ID's as 
```
problem_group:problem1
problem_group:problem2
. . .
```
The rendering of that colon was a problem because a link would stop there and wouldn't function. That's why the replacement was necessary, also only dashes and colons are likely to be present in these ID's and therefor only the two are replaced, but I've left it open to add additional characters if they are found in the future.
For more context, please checkout the discussion in the comments here: https://openedx.atlassian.net/browse/TNL-2896

Also, I don't understand the "comment should be on its own line" comment.